File: hisi-dev-tool/src/main/java/com/huawei/hisi/project/remote/model/RemoteProject.java
package com.huawei.hisi.project.remote.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RemoteProject {
    private Long id;
    private String name;
    private String gitUrl;
    private String username;
    private String encryptedPassword;
    @Builder.Default
    private String branch = "main";
    private String localPath;
    @Builder.Default
    private String cloneStatus = "PENDING";
    private String cloneError;
    private Long lastSyncAt;
    private Long createdAt;

    // Auth type fields
    @Builder.Default
    private String authType = "PASSWORD"; // PASSWORD / SSH_KEY / TOKEN
    private String sshKeyPath; // SSH private key file path
    private String encryptedToken; // AES encrypted token

    // Group association
    private Long groupId;
}

File: hisi-dev-tool-frontend/src/types/remote-project.ts
export interface RemoteProject {
  id: number
  name: string
  gitUrl: string
  username: string
  branch: string
  localPath: string  // Project name slug (e.g., "activity-management-service")
  fullPath: string   // Complete physical path where project is cloned (e.g., "D:/codeknowledge/remote-repos/activity-management-service")
  cloneStatus: 'PENDING' | 'CLONING' | 'CLONED' | 'FAILED'
  cloneError: string | null
  lastSyncAt: number | null
  authType: 'PASSWORD' | 'SSH_KEY' | 'TOKEN'
  sshKeyPath?: string
  groupId?: string
  groupName?: string
}

export interface CreateRemoteProjectRequest {
  name: string
  gitUrl: string
  username?: string
  password?: string
  branch?: string
  authType?: 'PASSWORD' | 'SSH_KEY' | 'TOKEN'
  sshKeyPath?: string
  token?: string
  groupId?: string
}

export interface UpdateRemoteProjectRequest {
  name: string
  gitUrl: string
  username?: string
  password?: string
  branch?: string
  authType?: 'PASSWORD' | 'SSH_KEY' | 'TOKEN'
  sshKeyPath?: string
  token?: string
  groupId?: string
}

export interface KgSchedule {
  id: number
  projectPath: string
  cronExpression: string
  buildMode: 'INCREMENTAL' | 'REUSE' | 'WIPE'
  enabled: boolean
  gitPullEnabled: boolean
  branch: string
  refreshDescription: boolean
  refreshArchitecture: boolean
  lastRunAt: number | null
  nextRunAt: number | null
}

export interface CreateKgScheduleRequest {
  projectPath: string
  cronExpression: string
  buildMode: 'INCREMENTAL' | 'REUSE' | 'WIPE'
  gitPullEnabled?: boolean
  branch?: string
  refreshDescription?: boolean
  refreshArchitecture?: boolean
}

export interface UpdateKgScheduleRequest {
  projectPath: string
  cronExpression: string
  buildMode: 'INCREMENTAL' | 'REUSE' | 'WIPE'
  enabled: boolean
  gitPullEnabled: boolean
  branch: string
  refreshDescription: boolean
  refreshArchitecture: boolean
}
